crux-bench/src/crux/bench.clj

```diff
@@ -104,22 +106,145 @@
 (defmacro with-bench-ns [bench-ns & body]
   `(with-bench-ns* ~bench-ns (fn [] ~@body)))
 
-(defn with-node* [f]
+(defprotocol GetNode
+  (give [this]))
+
+(defn nodes [data-dir] {;"standalone-lmdb"
```

Comment:
few pointers about this:
* aware that embedded kafka might throw a spanner in the works here, but is there any way we can represent this more as data so that we don't have to create a new protocol + reify it?
* guessing you've considered this, but I've missed it - is there any way we can avoid the permutation explosion here?
* formatting nit: if a `def`/`defn` is more than one line, can we have it down a line?